astropy/time/formats.py

```diff
@@ -314,6 +322,10 @@ def _check_val_type(self, val1, val2):
                 f"Input values for {self.name} class must be finite doubles"
             )
 
+    def _check_val_type(self, val1, val2):
+        """Input value validation, typically overridden by derived classes."""
+        self._check_finite_vals(val1, val2)
```

Comment:
For me it is a bit more obvious to move the decision logic out to the main method instead of using a short-circuit in `_check_finite_vals`.
```suggestion
        if self._check_finite:
            self._check_finite_vals(val1, val2)
```

Reply:
easy enough.